### PR TITLE
Log more dpo metrics

### PR DESCRIPTION
**Pull request by maxjeblick**
This PR logs more metrics for DPO training (in particular cross entropy loss). 

I refactored train.py a bit, it is now also possible to automatically log additional metrics/losses by adding them to the model's output dictionary using `additional_log_` prefix (e.g. `outputs["additional_log_chosen_rewards"] = ...`)

Related to #594

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -85,12 +84,13 @@ def run_eval(
     Returns:
         Validation loss
     """
-
+    torch.inference_mode(mode=True)
     with torch.no_grad():
```
Comment (pascal-pfeiffer) [inline train.py:88]: `no_grad` is probably redundant now, but we can leave it in there. 

I remember we had some issues with inference_mode a while back. Was it a necessary change? Otherwise, i'd stick with the old solution.

**Review by pascal-pfeiffer — CHANGES_REQUESTED** [review]
Thanks a lot @maxjeblick 
Super helpful to have more metrics logged. Ideally, we would also want to show them in the UI, but having them on neptune and in the log files, is already much better.

Please have another look on the path change for the gpt_templates. I believe this slipped into your PR somehow. Otherwise, looks good to me!

File: llm_studio/python_configs/text_causal_language_modeling_config.py [hunk 2]
```diff
@@ -318,8 +318,11 @@ def __post_init__(self):
             ),
             allow_custom=True,
         )
+        prompt_template_directory = os.path.join(
+            os.path.dirname(__file__), "../../prompts"
+        )
         self._possible_values["metric_gpt_template"] = possible_values.String(
-            values=(f.split(".")[0] for f in os.listdir("prompts"))
+            values=(f.split(".")[0] for f in prompt_template_directory)
```
[the remark this answers is not in the document]
Comment (maxjeblick) [reply llm_studio/python_configs/text_causal_language_modeling_config.py:325]: Thanks, will check. Reason for the change is to resolve `prompts` directory w.r.t. `llm_studio/python_configs/text_causal_language_modeling_config.py` to be able to run tests with root directory == directory where test is running (this is the default in PyCharm when a test is started within the code editor).

File: train.py (hunk 1 above)
Comment (maxjeblick) [reply train.py:88]: Thanks for the catch, inference mode can probably be removed. 
Slipped in from https://github.com/h2oai/h2o-llmstudio/blob/main/train.py#L102 where it is actually not needed.

**Review by pascal-pfeiffer — APPROVED** [review]
Thanks again, lgtm!
